### ISSUE TITLE
Hyperparameter fixes

### ISSUE THREAD

**Issue by SamuelMarks** [SamuelMarks/ml-glaucoma#1]
This issue tracks the implementation + bug fixing of hyperparameter tasks.

Hyperparameter | Option† |  Option† | Option†
------------ | -------------| -------- | --------
num_classes | 1 | 2
activation | softmax | any [one] [`keras.activiations`](https://github.com/keras-team/keras/blob/master/keras/activations.py) function
epoch | number > 0
color_mode | grayscale (xor 1) | rgb (xor 3)
transfer_model | dc0 | any [one] [keras.applications](https://github.com/keras-team/keras/tree/master/keras/applications) model
architecture | `unet`; for U-Net architecture
metrics | `precision_recall` | `binary_true_positives` | any [one] [streaming metrics (particularly: AUC)](https://github.com/keras-team/keras/issues/8657#issuecomment-456520303)
loss | `weighted_categorical_crossentropy` | `binary_segmentation_recall` | any [one] [`keras.losses`](https://github.com/keras-team/keras/blob/master/keras/losses.py) function
optimizers | any [one] [`keras.optimizers`](https://github.com/keras-team/keras/blob/master/keras/optimizers.py) function
tensorboard_log_dir | location for [tensorboard](https://github.com/tensorflow/tensorboard) to serve analyzable files from
dropout | number to configure amount of dropout in non transfer learning model
pixels | one number showing the dimensions, e.g.: 5 for a 25px image (5×5)
write_images | boolean parameter for [`TensorBoard`](https://github.com/keras-team/keras/blob/a139716/keras/callbacks.py#L942)
model_name | Filename for h5 trained model file
batch_size | number > 0
preprocess_to | Filename to save h5 file of dataset, following preprocessing

† one at a time

There are others in [README.md](https://github.com/SamuelMarks/ml-glaucoma/blob/88174e3/README.md), but the remainder should work unedited. Almost all the above should also work unedited.

You can find the implementation as it currently stands, in a [single function named `run`](https://github.com/SamuelMarks/ml-glaucoma/blob/88174e3958f67d95a4a80c5c66011a832ab9502b/ml_glaucoma/CNN/bmes_cnn.py#L104).

**Comment by jackd** [comment]
In branch `jackd-v2`:
## num_classes
Implicitly defined by `Problem.output_spec`.

## epoch
`--epochs`

## color mode
`--gray` and `--gray_on_disk` can be used to either map rgb images stored on disk to grayscale or construct a separate grayscale dataset on disk respectively.

## models / architecture / activation / dropout / pixels
Configurable via gin. Not entirely sure what "pixels" means, but should be easy enough to add support.

## Metrics
Support for:
* BinaryAccuracy
* AUC
* Precision (via --precision_thresholds)
* Recall (via --recall_thresholds)

## Optimizers
All tf.keras.optimizers supported

## tensorboard_log_dir / write_images
`-tb`, or defaults to `--model_dir`
`--write_images`

## model_name
Supports `--model_dir` - model files will be saved as `model_dir/'model-{epoch:04d}.h5'` - is that acceptable?

## batch_size
`-b`, `--batch_size`

## preprocess_to
tfrecords are saved at `data_dir/builder_name/version`, where `--data_dir` is a CL arg and `builder_name` is the name of the tfds builder, e.g. `refuge/r256-256-rgb`
